src/git-source-provider.ts

```diff
@@ -0,0 +1,222 @@
+import * as core from '@actions/core'
+import * as coreCommand from '@actions/core/lib/command'
+import * as fs from 'fs'
+import * as fsHelper from './fs-helper'
+import * as gitCommandManager from './git-command-manager'
+import * as io from '@actions/io'
+import * as path from 'path'
+import * as refHelper from './ref-helper'
+import {IGitCommandManager} from './git-command-manager'
+
+const authConfigKey = `http.https://github.com/.extraheader`
+
+export interface ISourceSettings {
+  repositoryPath: string
+  repositoryOwner: string
+  repositoryName: string
+  ref: string
+  commit: string
+  clean: boolean
+  fetchDepth: number
+  lfs: boolean
+  accessToken: string
+}
+
+export async function getSource(settings: ISourceSettings): Promise<void> {
+  core.info(
+    `Syncing repository: ${settings.repositoryOwner}/${settings.repositoryName}`
+  )
+  const repositoryUrl = `https://github.com/${encodeURIComponent(
+    settings.repositoryOwner
+  )}/${encodeURIComponent(settings.repositoryName)}`
+
+  // Remove conflicting file path
+  if (fsHelper.fileExistsSync(settings.repositoryPath)) {
+    await io.rmRF(settings.repositoryPath)
+  }
+
+  // Create directory
+  let isExisting = true
+  if (!fsHelper.directoryExistsSync(settings.repositoryPath)) {
+    isExisting = false
+    await io.mkdirP(settings.repositoryPath)
+  }
+
+  // Git command manager
+  core.info(`Working directory is '${settings.repositoryPath}'`)
+  const git = await gitCommandManager.CreateCommandManager(
+    settings.repositoryPath,
+    settings.lfs
+  )
+
+  // Try prepare existing directory, otherwise recreate
+  if (
+    isExisting &&
+    !(await tryPrepareExistingDirectory(
+      git,
+      settings.repositoryPath,
+      repositoryUrl,
+      settings.clean
+    ))
+  ) {
+    await io.rmRF(settings.repositoryPath)
+    await io.mkdirP(settings.repositoryPath)
+  }
+
+  // Initialize the repository
+  if (
+    !fsHelper.directoryExistsSync(path.join(settings.repositoryPath, '.git'))
+  ) {
+    await git.init()
+    await git.remoteAdd('origin', repositoryUrl)
+  }
+
+  // Disable automatic garbage collection
+  if (!(await git.tryDisableAutomaticGarbageCollection())) {
+    core.warning(
+      `Unable to turn off git automatic garbage collection. The git fetch operation may trigger garbage collection and cause a delay.`
+    )
+  }
+
+  // Remove possible previous extraheader
+  await removeGitConfig(git, authConfigKey)
+
+  // Add extraheader (auth)
+  const base64Credentials = Buffer.from(
+    `x-access-token:${settings.accessToken}`,
+    'utf8'
+  ).toString('base64')
+  core.setSecret(base64Credentials)
+  const authConfigValue = `AUTHORIZATION: basic ${base64Credentials}`
+  await git.config(authConfigKey, authConfigValue)
+
+  // LFS install
+  if (settings.lfs) {
+    await git.lfsInstall()
```

Comment:
do we need to do something similar to https://github.com/actions/runner/blob/master/src/Runner.Plugins/Repository/v1.0/GitSourceProvider.cs#L146

Reply:
it's in git-command-manager.ts in the initializeCommandManager function